protos/sigstore_rekor.proto

```diff
@@ -0,0 +1,109 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.rekor.v1;
+
+option go_package = "github.com/sigstore/proto/rekor/v1";
+option java_package = "dev.sigstore.proto.rekor.v1";
+option java_multiple_files = true;
+option java_outer_classname = "RekorProto";
+
+// KindVersion contains the entry's kind and api version.
+message KindVersion {
+        // Kind is the type of entry being stored in the log.
+        // See here for a list: https://github.com/sigstore/rekor/tree/main/pkg/types
+        string kind = 1;
+        // The specific api version of the type.
+        string version = 2;
+}
+
+// The checkpoint contains a signature of the tree head (root hash),
+// size of the tree, the transparency log's unique identifier (log ID),
+// hostname and the current time.
+// The result is a string, the format is described here
+// https://github.com/transparency-dev/formats/blob/main/log/README.md
+// The details are here https://github.com/sigstore/rekor/blob/main/pkg/util/signed_note.go#L114.
+// The signature has the same format as
+// TransparencyLogEntry.inclusion_proof. See below for more details.
+message Checkpoint {
+        string envelope = 1;
+}
+
+// InclusionProof is the proof returned from the transparency log. Can
+// be used for on line verification against the log.
+message InclusionProof {
+        // The index of the entry in the log.
+        int64 log_index = 1;
+        // The hash digest stored at the root of the merkle tree at the time
+        // the proof was generated.
+        bytes root_hash = 2;
+        // The size of the merkle tree at the time the proof was generated.
+        int64 tree_size = 3;
+        // A list of hashes required to compute the inclusion proof, sorted
+        // in order from leaf to root.
+        repeated bytes hashes = 4;
+        // Signature of the tree head, as of the time of this proof was
+        // generated. See above info on 'Checkpoint' for more details.
+        Checkpoint checkpoint = 5;
+}
+
+// The inclusion promise is calculated by Rekor. It's calculated as a
+// signature over a canonical JSON serialization of the persisted entry, the
+// log ID, log index and the integration timestamp.
+// See https://github.com/sigstore/rekor/blob/main/pkg/api/entries.go#L54
+// The format of the signature depends on the transparency log's public key.
+// If the signature algorithm requires a hash function and/or a signature
+// scheme (e.g. RSA) those has to be retrieved out-of-band from the log's
+// operators, together with the public key.
+// This is primarily used to verify the integration timestamp's value.
+message InclusionPromise {
+        bytes signed_entry_timestamp = 1;
+}
+
+// LogId captures the identity of a transparceny log.
+message LogId {
```

Comment:
nit: `LogId` or `LogID`?

Reply:
I think `LogId`, using `ID` over `Id` seems more a like go thing to me.

Reply:
Sg